updated according to comments from 09.10.2024 from mentor

YauheniyaH opened the pull request.

YauheniyaH committed:
updated according to comments from 09.10.2024 from mentor

ADrema commented on src/main/java/pageobject_model/page/CloudGoogleComputeEnginePage.java:
```diff
@@ -7,51 +7,70 @@
+import static java.lang.String.format;
+import static java.util.logging.Level.INFO;
+import static org.openqa.selenium.OutputType.FILE;
+
```
one empty line is enough

YauheniyaH replied:
done

ADrema commented on src/main/java/pageobject_model/test/WebDriverCloudGoogleTest.java:
```diff
@@ -35,14 +35,19 @@ public void browserSetup() throws IOException {
         computeEnginePage.inputNumberOfInstances("4");
         computeEnginePage.optionalPopUpClose();
+
         computeEnginePage.selectDropdownValue("OS","'Free: Debian, CentOS, CoreOS, Ubuntu or BYOL (Bring Your Own License)'");
```
OS, machineType, machineType can be moved into separate class as Dropdown values

YauheniyaH replied:
will move to enum inside DropDownObject class

ADrema commented on src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:
```diff
@@ -14,27 +16,41 @@
 
+    @Test
+    public void seleniumGridTest() {
+        assertEquals("Compute Engine", computeEnginePage.pageTitle.getText());
```
please open the page inside the test

YauheniyaH replied:
```suggestion
        computeEnginePage.openPage();
        assertEquals("Compute Engine", computeEnginePage.pageTitle.getText());
```

YauheniyaH replied:
will be added in next commit like suggested

ADrema commented on src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:
```diff
@@ -14,27 +16,41 @@
+    private CloudGoogleComputeEnginePage computeEnginePage;
+
+    @BeforeMethod(alwaysRun = true)
+    public void browserSetup() throws IOException, URISyntaxException {
```
please move current tests to selenium Grid

YauheniyaH replied:
added to @BeforeMethod driver creation steps using RemoteWebDriver and DesiredCapabilities

ADrema commented on src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:
```diff
@@ -14,27 +16,41 @@
+    @AfterMethod(alwaysRun = true)
+    public void browserTearDown() {
         driver.quit();
+        driver = null;
```
Do you really need it?

YauheniyaH replied:
This helps ensure that the browser session is completely closed and the memory is released.

YauheniyaH committed:
1. Try to implement element click with JS Executor -  implemented in CloudGoogleComputeEnginePage.optionalPopUpClose
2. one empty line is enough - done
3. OS, machineType, machineType can be moved into separate class as Dropdown values - moved to enum inside DropDownObject class
4. please open the page inside the test - done
5. please move current tests to selenium Grid - added to @BeforeMethod driver creation steps using RemoteWebDriver and DesiredCapabilities for WebDriverCloudGoogleTest
6. Do you really need it? driver=null - This helps ensure that the browser session is completely closed and the memory is released.
